### PR TITLE
str.format

### PR DESCRIPTION
**Pull request by jstoebel**
when complete, this will implement `str.format`

### CODE REVIEW

**Comment by wolfv** [comment]
Maybe it also makes sense to incorporate some tests from `brython`: https://github.com/brython-dev/brython/blob/master/www/tests/test_string_format.py

**Comment by jstoebel** [comment]
hmmm, now there's a bunch of failing tests and I'm not sure what I did to fail them. Just out of curiosity I looked at the first one.

```
======================================================================
FAIL: test_abs_not_implemented (tests.builtins.test_abs.AbsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/stoebelj/batavia/tests/builtins/test_abs.py", line 14, in test_abs_not_implemented
    """, run_in_function=False)
  File "/Users/stoebelj/batavia/tests/utils.py", line 446, in assertCodeExecution
    self.assertEqual(js_out, py_out, context)
AssertionError: '### EXCEPTION ###\nTypeError: TypeError:[48 chars]:6\n' != "### EXCEPTION ###\nTypeError: bad operan[43 chars]:6\n"
  ### EXCEPTION ###
- TypeError: TypeError: types.isinstance is not a function
+ TypeError: bad operand type for abs(): 'NotAbsLike'
      test.py:6
 : Global context
```

inside `abs.js` I moved `var types = require('../types')` inside of the `abs` function and the test passes.

Any ideas where I've buggered things up? I'm stumped.

**Comment by jstoebel** [comment]
UPDATE: [This PR](https://github.com/pybee/batavia/pull/582) passes. it touches three files and it appears that the edits I made to those same files in _this_ PR are effectively the same. If anyone ha any ideas on where the problem might lie I'd love to know.

**Comment by jstoebel** [comment]
I've rollback changes to `hash` and `types` and I'm still getting many errors. I'm stumped as to why.

File: tests/builtins/test_js2py.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from .. utils import TranspileTestCase
```
Comment (freakboy3742) [inline tests/builtins/test_js2py.py]: This file is redundant; I think it's an artefact of an earlier merge with the j2py patch.

Comment (jstoebel) [inline tests/builtins/test_js2py.py]: file removed!

**Pull request merged**